src/olympia/devhub/tasks.py

```diff
@@ -624,13 +625,24 @@ def resize_icon(source, dest_folder, target_sizes, **kw):
     """Resizes addon icons."""
     log.info('[1@None] Resizing icon: %s' % dest_folder)
     try:
+        # Resize in every size we want.
         dest_file = None
         for size in target_sizes:
             dest_file = '%s-%s.png' % (dest_folder, size)
             resize_image(source, dest_file, (size, size))
+
+        # Store the original hash, we'll return it to update the corresponding
+        # add-on.
+        with open(source) as fd:
+            icon_hash = hashlib.md5(fd.read()).hexdigest()[:8]
```

Comment:
I'm not sure I'm a fan of this even though there's a good chance that this is unique enough and we have the icon-name as another factor. Maybe just a comment about the logic here and why the first bits are enough would suffice. As far as I know md5 hashes are relatively evenly distributed so we probably won't run into problems for many thousand icons to come. I wouldn't go much higher than that though. I wonder why we have that limitation? Bandwidth, storage?

Reply:
I could go further but I don't really see the point, IMHO that wastes storage space - this will only be an issue if the same add-on changes their icon and somehow the hexadecimal representation of the md5 of their new icon has the same first 8 chars. Icon changes are rare enough in the first place... ending up with such a similar md5 seems extremely unlikely.

Reply:
I do agree, let's document that though.